src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java

```diff
@@ -342,6 +342,7 @@ private synchronized void initializeSelfTests() {
         selfTestSuite.addSelfTest(EvpHmac.SHA1.SELF_TEST);
         selfTestSuite.addSelfTest(EvpHmac.MD5.SELF_TEST);
         selfTestSuite.addSelfTest(LibCryptoRng.SPI.SELF_TEST);
+        selfTestSuite.addSelfTest(SelfTestSuite.AWS_LC_SELF_TESTS);
```

Comment:
This is all hypothetical, but shouldn't we run AWS-LC's self tests first before ACCP's own internal self tests? 

If an algorithm in AWS-LC will fail AWS-LC's self tests (and presumably our ACCP self tests as well), I'd rather that an ACCP user see the AWS-LC self-test failure first so that they know where to start debugging.

Reply:
That makes sense. The self-tests are stored in a HashMap and the order they get inserted does not determine the order they get executed and reported. I can change the underlying data structure to ensure the order of execution matches the order of insertion.